accesshandler/pkg/providers/providers.go

```diff
@@ -20,6 +20,12 @@ type Accessor interface {
 	Revoke(ctx context.Context, subject string, args []byte, grantID string) error
 }
 
+// If a provider implements RequiresAccessToken
+// It indicates that an access token should be generated for use by this provider.
+type RequiresAccessToken interface {
+	RequiresAccessToken()
```

Comment:
an empty function like this looks a bit weird - let's make this return a boolean. For access providers which use this we can return `true`.
```suggestion
	RequiresAccessToken() bool
```